integration_tests/base_routes.py

```diff
@@ -449,6 +449,35 @@ async def async_template_render():
     return template
 
 
+# add route and it's function for url_for for template for integration testing
+@app.get("/sync/url_for")
+def sync_url_for():
+    # Generate URL for the "/about" route
+    about_url = url_for("/about")
+
+    # Generate URL for the "/contact" route with query parameters
+    contact_url = url_for("/contact", name="John", email="john@example.com")
+
+    # Generate URL for the static resource "styles.css"
+    static_url = url_for("static", filename="styles.css")
+
+    return f"About URL: {about_url}, Contact URL: {contact_url}, Static URL: {static_url}"
+
+
+@app.get("/async/url_for")
+async def async_url_for():
+    # Generate URL for the "/about" route
+    about_url = url_for("/about")
+
+    # Generate URL for the "/contact" route with query parameters
+    contact_url = url_for("/contact", name="John", email="john@example.com")
+
+    # Generate URL for the static resource "styles.css"
+    static_url = url_for("static", filename="styles.css")
+
+    return f"About URL: {about_url}, Contact URL: {contact_url}, Static URL: {static_url}"
+
```

Comment:
@carrycooldude , the `url_for` function is injected in the template rendering process. Hence, it should be used in the `url_for.html` This is again a unit test, which is not what we want.